src/PrestaShopBundle/Controller/Admin/FrameworkBundleAdminController.php

```diff
@@ -28,24 +28,80 @@
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
 use PrestaShop\PrestaShop\Core\Help\Documentation;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
 use PrestaShop\PrestaShop\Core\Localization\Locale;
 use PrestaShop\PrestaShop\Core\Localization\Locale\Repository as LocaleRepository;
 use PrestaShop\PrestaShop\Core\Module\Exception\ModuleErrorInterface;
 use PrestaShop\PrestaShop\Core\Security\Permission;
+use Psr\Container\ContainerInterface;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\Form\FormError;
 use Symfony\Component\Form\FormInterface;
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Contracts\Translation\TranslatorInterface;
 
 /**
  * Extends The Symfony framework bundle controller to add common functions for PrestaShop needs.
  */
 class FrameworkBundleAdminController extends AbstractController
 {
-    public const PRESTASHOP_CORE_CONTROLLERS_TAG = 'prestashop.core.controllers';
+    /**
+     * The Symfony global service container was used inside all controllers and replaced the small container injected by default in recent Symfony versions.
+     * It is temporary and will be removed in the future, When all "this->get()" calls will be replaced by proper dependency injection.
+     * This also will avoid uses of public services inside controllers. To cleanup dependency injection.
+     *
+     * @see \PrestaShopBundle\DependencyInjection\Compiler\ContainerInjectionPass
+     *
+     * @var ContainerInterface
+     *
+     * @internal
+     */
+    private $globalContainer;
+
+    public function setGlobalContainer(ContainerInterface $container)
+    {
+        $this->globalContainer = $container;
+    }
+
+    /**
+     * @see self::$globalContainer
+     *
+     * Returns the service with the given id from the controllers container or the global container if not found. Throws a deprecation if needed.
+     */
+    protected function get(string $id)
+    {
+        if ($this->container->has($id)) {
+            return $this->container->get($id);
+        }
+
+        @trigger_error(sprintf('Fetching the "%s" service directly from the container is deprecated since Prestashop 9. Inject it as a dependency instead.', $id), E_USER_DEPRECATED);
+
+        return $this->globalContainer->get($id);
+    }
+
+    protected function getGrid(string $grid, SearchCriteriaInterface $searchCriteria): GridInterface
+    {
+        return $this->container->get('grid_manager')->getGridFactory($grid)->getGrid($searchCriteria);
+    }
+
+    /**
+     * List of services commonly used in all PrestaShop controllers.
+     */
+    public static function getSubscribedServices(): array
+    {
+        // Use all symfony services as default, and then add prestashop services.
+        return parent::getSubscribedServices() + [
+            'grid_manager' => GridFactoryManagerInterface::class,
```

Comment:
```suggestion
            GridFactoryManagerInterface::class => GridFactoryManagerInterface::class,
```
also wdyt maybe its worth to rename it to GridCollector? (manager sounds so wast it could do anything)

Reply:
Not a big fan of the gridcollector, as it makes me think about datacollectors we use in Symfony. Maybe `GridContainer` to keep it simple ~stupid~ 😄 . Because in fact, this is just a container.